AppCenterDataStorage/AppCenterDataStorageTests/MSDBDocumentStoreTests.m

```diff
@@ -202,6 +200,109 @@ - (void)testReadUserDocumentFromLocalDatabaseNotFound {
+
+  // When
+
+  // Read once and update the TTL to expire immediately, then read again to ensure the document is expired.
+  [self.sut readWithToken:self.appToken documentId:documentId documentType:[MSMockDocument class] readOptions:readOptions];
+
+  // Then
+  OCMVerify([self.sut deleteWithToken:self.appToken documentId:documentId]);
+}
+
+- (void)testReadFromLocalDatabaseCorrectlyUpdatesTTLToInfinity {
+
+  // If
+  NSString *documentId = @"12829";
+  NSString *eTag = @"398";
+  NSString *jsonString = @"{ \"document\": {\"key\": \"value\"}}";
+  NSString *pendingOperation = kMSPendingOperationReplace;
+  [self addJsonStringToTable:jsonString
+                        eTag:eTag
+                   partition:self.appToken.partition
+                  documentId:documentId
+            pendingOperation:pendingOperation
+              expirationTime:(long)[[NSDate dateWithTimeIntervalSinceNow:1000000] timeIntervalSince1970]];
+  MSReadOptions *readOptions = [[MSReadOptions alloc] initWithDeviceTimeToLive:MSDataStoreTimeToLiveInfinite];
+
+  // When
+
+  // Read once and update the TTL to expire immediately, then read again to ensure the document is expired.
+  [self.sut readWithToken:self.appToken documentId:documentId documentType:[MSMockDocument class] readOptions:readOptions];
+
+  // Then
+  long expirationTime = [self expirationTimeWithToken:self.appToken documentId:documentId];
+  XCTAssertEqual(expirationTime, MSDataStoreTimeToLiveInfinite);
+}
+
+- (void)testReadFromLocalDatabaseCorrectlyUpdatesTTL {
+
+  // If
+  NSString *documentId = @"12829";
+  NSString *eTag = @"398";
+  NSString *jsonString = @"{ \"document\": {\"key\": \"value\"}}";
+  NSString *pendingOperation = kMSPendingOperationReplace;
+  [self addJsonStringToTable:jsonString
+                        eTag:eTag
+                   partition:self.appToken.partition
+                  documentId:documentId
+            pendingOperation:pendingOperation
+              expirationTime:(long)[[NSDate dateWithTimeIntervalSinceNow:1000000] timeIntervalSince1970]];
+  long timeToLive = 158493;
+  MSReadOptions *readOptions = [[MSReadOptions alloc] initWithDeviceTimeToLive:timeToLive];
+
+  // Mock NSDate to "freeze" time.
+  NSTimeInterval timeSinceReferenceDate = NSDate.timeIntervalSinceReferenceDate;
+  id nsdateMock = OCMClassMock([NSDate class]);
+  OCMStub(ClassMethod([nsdateMock timeIntervalSinceReferenceDate])).andReturn(timeSinceReferenceDate);
+
+  // When
+
+  // Read once and update the TTL to expire immediately, then read again to ensure the document is expired.
+  [self.sut readWithToken:self.appToken documentId:documentId documentType:[MSMockDocument class] readOptions:readOptions];
+
+  // Then
+  long expirationTime = [self expirationTimeWithToken:self.appToken documentId:documentId];
+  XCTAssertEqual(expirationTime, (long)(timeToLive + NSTimeIntervalSince1970 + timeSinceReferenceDate));
+}
+
+- (void)testUpsertReplacesCorrectlyInAppStorage {
```

Comment:
This test is already in #1499 - I am not sure if that PR is separate or if this is just a larger version of the same PR